grails-app/services/de/iteratec/osm/system/MissingJobResultCheckService.groovy

```diff
@@ -0,0 +1,103 @@
+package de.iteratec.osm.system
+
+import de.iteratec.osm.measurement.environment.wptserver.JobResultPersisterService
+import de.iteratec.osm.measurement.schedule.Job
+import de.iteratec.osm.result.JobResult
+import de.iteratec.osm.result.JobResultStatus
+import grails.gorm.transactions.Transactional
+import org.joda.time.DateTime
+import org.quartz.CronExpression
+
+@Transactional
+class MissingJobResultCheckService {
+    static final int FIRST_CHECK_DAYS_AGO = 1
+    static final long SCHEDULE_THRESHOLD_MINUTES = 3
+
+    JobResultPersisterService jobResultPersisterService
+
+    def fillMissingJobResults() {
+        Date latestCheckDate = getLatestCheck()?.date
+        int missingResults = 0
+        if(latestCheckDate) {
+            missingResults = fillMissingJobResultsSinceDate( new DateTime(latestCheckDate).minusMinutes(SCHEDULE_THRESHOLD_MINUTES.toInteger()))
+            new MissingJobResultCheck(date: new Date(), missingResults: missingResults).save()
+        }
+        else {
+            DateTime dateDaysAgo =  new DateTime().minusDays(FIRST_CHECK_DAYS_AGO)
+            missingResults = fillMissingJobResultsSinceDate( dateDaysAgo )
+            new MissingJobResultCheck(date: new Date(), missingResults: missingResults).save()
```

Comment:
You can avoid some logic duplication by only defining a `startDate` (or similar) in the if/else branch and executing the rest after the branch. This logic can be as simple as `DateTime startDate = latestCheckDate ? new DateTime(latestCheckDate).minusMinutes(SCHEDULE_THRESHOLD_MINUTES) : new DateTime().minusDays(FIRST_CHECK_DAYS_AGO)`